### PR TITLE
Separate metering for simple composites

### PR DESCRIPTION
**Pull request by dsainati1**
With the changes in https://github.com/onflow/cadence/pull/1591, it no longer makes sense to meter `SimpleCompositeValue`s (which do not have an underlying atree) like regular `CompositeValue`s, which do. 
______

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: runtime/common/metering.go [hunk 1]
```diff
@@ -197,6 +197,13 @@ func NewCompositeMemoryUsages(size uint64, element_size uint) (MemoryUsage, Memo
 	}, leaves, branches
 }
 
+func NewSimpleCompositeMemoryUsage(length int) MemoryUsage {
+	return MemoryUsage{
+		Kind:   MemoryKindSimpleComposite,
+		Amount: uint64(length) + 1, // +1 to account for empty strings
```
Comment (SupunS) [inline runtime/common/metering.go:203]: Why is an additional +1 needed?

File: runtime/interpreter/simplecompositevalue.go [hunk 2]
```diff
@@ -56,10 +56,8 @@ func NewSimpleCompositeValue(
 	stringer func(SeenReferences) string,
 ) *SimpleCompositeValue {
 
-	baseUsage, dataUsage, metaDataUsage := common.NewCompositeMemoryUsages(uint64(len(fields)), 0)
-	common.UseMemory(inter, baseUsage)
-	common.UseMemory(inter, dataUsage)
-	common.UseMemory(inter, metaDataUsage)
+	common.UseMemory(inter, common.NewConstantMemoryUsage(common.MemoryKindCompositeBase))
```
Comment (SupunS) [inline runtime/interpreter/simplecompositevalue.go:59]: Is the base same for both composites and simple-composites? The fields of the two go-structs are different, so the weights also need to be different.

Also, the base memory-usage (i.e: `common.NewConstantMemoryUsage(common.MemoryKindCompositeBase)`) can be pre computed / 
 computed only once

File: docs/language/accounts.mdx [hunk 3]
```diff
@@ -88,6 +88,19 @@ to the `prepare` phase of the transaction.
...
+      // Adds a public key to the account.
+      // The public key must be encoded together with their signature algorithm, hashing algorithm and weight.
+      // This method is currently deprecated and is available only for the backward compatibility.
+      // `keys.add` method can be use instead.
+      fun addPublicKey(_ publicKey: [UInt8])
+
+      // Revokes the key at the given index.
+      // This method is currently deprecated and is available only for the backward compatibility.
+      // `keys.revoke` method can be use instead.
+      fun removePublicKey(_ index: Int)
```
Comment (SupunS) [inline docs/language/accounts.mdx:102]: I guess these should go away if re-sync with the feature-branch?

Comment (dsainati1) [reply docs/language/accounts.mdx:102]: Ah yea, i will resync everything to fix this.

**Review by SupunS — APPROVED** [review]

**Pull request merged**